server/src/main/java/org/elasticsearch/cluster/coordination/Coordinator.java

```diff
@@ -670,6 +686,12 @@ public void onFailure(Exception e) {
         });
     }
 
+    /**
+     * Validates a request to join the new cluster. Runs on the candidate node running for election to master.
+     *
+     * @param joinRequest
+     * @param validateListener
```

Comment:
nit: I'd rather not have empty `@param` tags like this - we should either add some words here or just remove the tags.